### PR TITLE
JSON to PostgreSQL table

### PR DESCRIPTION
**Pull request by JakobMiksch**
Worker that reads a JSON from disk and inserts it into a specific table. 

This worker is not planned for production but rather serves as a proof of concept.

This is the entry in `docker-compose.yml`:
```yaml
  add-json-to-postgres-table:
    image: add-json-to-postgres-table:dev
    restart: unless-stopped
    volumes:
      - tmp-data:/home/data:Z
    depends_on:
      - rabbitmq
    environment:
      - RABBITHOST=${RABBITMQ_HOSTNAME}
      - RABBITUSER=${RABBITMQ_DEFAULT_USER}
      - RABBITPASS=${RABBITMQ_DEFAULT_PASS}
      - RESULTSQUEUE=${RABBITMQ_RESULTSQUEUE}
      - WORKERQUEUE=add-json-to-postgres-table
      - PGUSER=${POSTGRES_USER}
      - PGPASSWORD=${POSTGRES_PASSWORD}
      - PGHOST=postgres
      - PGPORT=5432
      - PGDATABASE=${POSTGRES_DB}
```

### CODE REVIEW

File: src/json-to-postgres-table/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  } catch (error) {
+    throw 'Reading JSON from disk failed'
+  }
+
+  try {
+    const client = new Client()
+    await client.connect()
+    let sql = `INSERT INTO ${table} (json, geom) VALUES (
+        '${jsonAsString}'::json,
+        ST_SetSRID(ST_MakePoint(7.2, 42.3), 4326)
```
Comment (weskamm) [inline src/json-to-postgres-table/index.js:72]: why do you use a st_makepoint call here? srid should be read from geojson in the end i think...

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "geoserver-node-client": "0.0.5",
     "node-fetch": "^2.6.1",
     "nodemailer": "^6.6.2",
+    "pg": "^8.7.3",
```
Comment (weskamm) [inline package.json:18]: please rebase and make sure all the old dependencies will dissappear here. You will now need to move the package.json and the lockfile to your new specific worker directory.

File: src/json-to-postgres-table/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM node:alpine
+
+COPY ./package.json ./package-lock.json /home/
```
Comment (weskamm) [inline src/json-to-postgres-table/Dockerfile:3]: these pathes will now need adjustment, see the other Dockerfiles for reference

**Review by weskamm — CHANGES_REQUESTED** [review]

**Comment by weskamm** [comment]
added comments about the restructuring that just happened. please adjust, else this is good to go

File: package.json (hunk 2 above)
Comment (JakobMiksch) [reply package.json]: alright. Done

File: src/json-to-postgres-table/Dockerfile (hunk 3 above)
Comment (JakobMiksch) [reply src/json-to-postgres-table/Dockerfile]: Thx. I updated it.

File: src/json-to-postgres-table/index.js (hunk 1 above)
Comment (JakobMiksch) [reply src/json-to-postgres-table/index.js]: I actually only had the geometry as an example. This worker is for the moment purely about reading a JSON and inserting it into a table. It should rather serve as demo, the geo-part is not important yet.

**Comment by JakobMiksch** [comment]
@weskamm I upgraded the PR to the latest changes in the repo. I addressed your notes. 
Meanwhile, I am not sure if we should merge this PR yet, because it depends on many other questions like database layout and which worker will actually create a table if necessary.

**Review by weskamm — APPROVED** [review]

**Comment by weskamm** [comment]
I would leave this up to you. You may also flag this with an appropriate title / tag and we can keep it open for demo purposes.

**Comment by JakobMiksch** [comment]
For the moment I would leave this PR as draft. Once we have more specificatations we can use it as blueprint or inspiration. I just want to avoid having an unused worker in our repo.

**Pull request closed without merging**